x/leverage/keeper/collateral.go

```diff
@@ -7,23 +7,28 @@ import (
 	"github.com/umee-network/umee/v2/x/leverage/types"
 )
 
-// burnCollateral removes some uTokens from an account's collateral and burns them. This occurs
-// during liquidations.
-func (k Keeper) burnCollateral(ctx sdk.Context, addr sdk.AccAddress, coin sdk.Coin) error {
-	err := k.setCollateralAmount(ctx, addr, k.GetCollateralAmount(ctx, addr, coin.Denom).Sub(coin))
+// liquidateCollateral burns utoken collateral and sends the base token reward to the liquidator.
+// This occurs during direct liquidation.
+func (k Keeper) liquidateCollateral(ctx sdk.Context, borrower, liquidator sdk.AccAddress, utoken, token sdk.Coin) error {
```

Comment:
The file `collateral.go` contains getters and setters related to uToken collateral, and other similarly small unexported functions. In particular, `burnCollateral` was meant to do three operations (burn, reduce recorded amount, reduce recorded supply) together so they can't get out of sync. I was planning to apply `burnCollateral` to the `amountFromCollateral` in `Withdraw` in the future as well.

I'm opposed to this function's refactor because it moves base token logic from `Liquidate` into this lower level uToken-only file.

Reply:
this function is also doing a basic operations. I don't see how `burnCollateral` make is better. In fact it makes the `Liquidate` function less clear. I can bring `burnCollateral` back and use it in `liquidateCollateral` helper function if that will help you.

Reply:
That should work - also, `liquidateCollateral` if created would probably go in `liquidate.go` while `burnCollateral` stays in `collateral.go`.

Reply:
Just applied proposed compromise (use burnCollateral inside liquidateCollateral helper function) in a commit. Didn't move anything between files for now. Approving now, but I'll leave merging to you so you can check if everything's good to go.